### ISSUE TITLE
Bug: CallExpression arguments indent is incorrect when first argument is an object

### ISSUE THREAD

**Issue by efyx** [eslint/eslint#17013]
### Environment

Node version: v18.12.1
npm version: v8.19.2
Local ESLint version: v8.26.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.2.0

### What parser are you using?

Default (Espree)

### What did you do?

```js
/* eslint indent: ["error", 2, {"CallExpression": {"arguments": 1}}] */

foo({
    foo: bar
  },
  arg2,
  arg3
);
```

### What did you expect to happen?

No lint errors reported since the formatting is actually correct

### What actually happened?

Eslint complains about the indentation level for the object properties as well as for the arguments of the function

![image](https://user-images.githubusercontent.com/80339/226870941-9ded0637-e151-40ec-b7a6-197b3983f3ea.png)

The "fixed" version by eslint looks like so (which seems to be incorrect)
```js
/* eslint indent: ["error", 2, {"CallExpression": {"arguments": 1}}] */
foo({
  foo: "bar"
},
arg2,
arg3
);

```

Note that the following works fine:

```js
/* eslint indent: ["error", 2, {"CallExpression": {"arguments": 1}}] */

// Works!
foo(arg1,
  arg2,
  arg3
);

// Also works
foo(
  {
    foo: "bar"
  },
  arg2,
  arg3
);

// Also works
foo(arg1,
  {
    foo: "bar",
  },
  arg3
);
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @efyx, thanks for the issue.

> The "fixed" version by eslint looks like so (which seems to be incorrect)
> 
> ```js
> /* eslint indent: ["error", 2, {"CallExpression": {"arguments": 1}}] */
> foo({
>   foo: "bar"
> },
> arg2,
> arg3
> );
> ```

What do you think the correct fix should be in this case?

**Comment by nzakas** [comment]
No followup, so closing.

**Comment by efyx** [comment]
@nzakas appologies, I totally missed your reply. Could you re-open?

> Hi @efyx, thanks for the issue.
> 
> > The "fixed" version by eslint looks like so (which seems to be incorrect)
> > ```js
> > /* eslint indent: ["error", 2, {"CallExpression": {"arguments": 1}}] */
> > foo({
> >   foo: "bar"
> > },
> > arg2,
> > arg3
> > );
> > ```
> 
> What do you think the correct fix should be in this case?

I was expecting the indentation to be as follow (arguments are indented instead of being at the start of the line which match the indentation style when the first argument isn't an object)
```
foo({
    foo: bar
  },
  arg2,
  arg3
);
```

**Comment by Rec0iL99** [comment]
@efyx not sure if it's a bug or indented behaviour. From my understanding, ESLint is treating the object as a whole entity like `arg2`. 

```js
foo({ bar:"bar" }, 
arg2,
arg3
)
```

This example causes the rule to report a spacing problem and the fix is:

```js
foo({ bar:"bar" }, 
  arg2,
  arg3
)
```

```js
foo({
  foo: "bar"
},
arg2,
arg3
);
```

I feel the correct fix here would be the following:

```js
foo(  {
    foo: "bar"
  },
  arg2,
  arg3
);
```

But this is not acceptable because of the 2 spaces at the start of the object's curly brackets.

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgMiwge1wiQ2FsbEV4cHJlc3Npb25cIjoge1wiYXJndW1lbnRzXCI6IDF9fV0gKi9cblxuZm9vKHsgYmFyOlwiYmFyXCIgfSwgXG4gIGFyZzIsXG4gIGFyZzNcbilcblxuZm9vKHtcbiAgZm9vOiBcImJhclwiXG59LFxuYXJnMixcbmFyZzNcbik7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6eyJlczYiOnRydWV9fX0=) 

I might be completely wrong though. 

cc @nzakas, @mdjermanovic 

Issues related to the indent rule are set at a low priority and we can't promise a fix. A new option would not be added because of the stylistic rules policy change state in the [README](https://github.com/eslint/eslint/#stylistic-rule-updates)

**Comment by sam3k** [comment]
Adding to this, adding extra spaces before the opening curly bracket does not trigger a lint warning/error:

```javascript
foo( { bar:"bar" }, 
  arg2,
  arg3
)
```

or:

```javascript
foo(   {
  foo: "bar"
},
arg2,
arg3
);
```

**Comment by nzakas** [comment]
I think it's fair to say that this is a bug given how the other examples behave. That said, the indent rule is a very complex one and fixing indentation bugs is fairly low on our priority list. If you'd like to submit a PR to try and fix this behavior, we'll happily review, but the team is unlikely to prioritize fixing this in the near future.

**Comment by nzakas** [comment]
Hi folks, I just wanted to let everyone know that the ESLint team has decided to deprecate all formatting rules, including indent, and recommend people use source code formatters like [Prettier](https://prettier.io) or [dprint](https://dprint.dev) instead. As a result, we are closing all outstanding issues related to these rules. For more information, please see https://github.com/eslint/eslint/issues/17522. Thanks.
